group-coordinator/src/test/java/org/apache/kafka/coordinator/group/Assertions.java

```diff
@@ -315,6 +317,57 @@ private static void assertApiMessageAndVersionEquals(
             actualValue.topicPartitions().sort(comparator);
 
             assertEquals(expectedValue, actualValue);
+        } else if (actual.message() instanceof ShareGroupPartitionMetadataValue) {
```

Comment:
This is a copy of the logic used for `ConsumerGroupPartitionMetadataValue` in this method. I will refactor how we implement those comparators in a follow-up PR. I am not happy with it.

Reply:
Here we go: https://github.com/apache/kafka/pull/17755.